File: 744-find-smallest-letter-greater-than-target/744-find-smallest-letter-greater-than-target.js
/**
 * @param {character[]} letters
 * @param {character} target
 * @return {character}
 */
var nextGreatestLetter = function(letters, target) {
    let start = 0;
    let end = letters.length-1;
    while(start<=end){
        let mid = Math.floor((start+end)/2);
        if(target<letters[mid]){
            end = mid -1;
        }
        else{
            start = mid +1
        }
    }
    return letters[start%letters.length]
};

File: 744-find-smallest-letter-greater-than-target/744-find-smallest-letter-greater-than-target.py
class Solution:
    def nextGreatestLetter(self, letters: List[str], target: str) -> str:
        
        
        start =0 
        end = len(letters)-1
        while(start<=end):
            mid = (start+end)//2
            if(target<letters[mid]):
                end = mid-1
            else:
                start = mid+1
        return letters[start%len(letters)]
            
   
